spec/models/user_spec.rb

```diff
@@ -0,0 +1,72 @@
+require "rails_helper"
+
+describe User, type: :model do
+  let(:new_user) { build(:user, :new) }
+  let(:transitioned_user) { build(:user, :transitioned) }
+  let(:rolled_over_user) { build(:user, :rolled_over, associated_with_accredited_body: true) }
+  let(:seen_accredited_body_features_user) { build(:user, :seen_accredited_body_new_features, associated_with_accredited_body: true) }
+
+  describe "transition state event" do
+    it "changes state from 'new' to 'transitioned'" do
+      stub_request(:post, "#{Settings.manage_backend.base_url}/api/v2/users").with(body: /"state":"transitioned"/)
```

Comment:
I think you're testing a bit more of the stack than you need to here, tbh. `aasm` comes with a rich set of [rspec matchers](https://github.com/aasm/aasm#rspec) which would avoid testing that we actually make a request.

Now that it's done this way we can leave it, but something to keep in mind for the future.